composeApp/src/commonMain/kotlin/presentation/feature/login/LoginScreen.kt

```diff
@@ -13,88 +13,78 @@ import androidx.compose.material.Scaffold
 import androidx.compose.material.Text
 import androidx.compose.material.TopAppBar
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.collectAsState
-import androidx.compose.runtime.getValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.text.input.PasswordVisualTransformation
 import androidx.compose.ui.text.style.TextAlign
-import cafe.adriel.voyager.core.screen.Screen
-import cafe.adriel.voyager.navigator.LocalNavigator
-import cafe.adriel.voyager.navigator.currentOrThrow
+import cafe.adriel.voyager.navigator.Navigator
 import presentation.components.appbutton.AppButton
 import presentation.components.apptextfield.AppTextField
-import presentation.feature.home.HomeScreen
+import presentation.feature.home.HomeScreenRoute
 import presentation.theme.Theme
 
-object LoginScreen : Screen {
-    @Composable
-    override fun Content() {
-        val viewModel = LoginViewModel()
-        val navigator = LocalNavigator.currentOrThrow
-        val state by viewModel.state.collectAsState()
-
-        Scaffold(
-            topBar = {
-                TopAppBar(
-                    title = {
-                        Text(
-                            text = "Login",
-                            fontWeight = FontWeight.Black,
-                            style = MaterialTheme.typography.h5,
-                            textAlign = TextAlign.Center,
-                            modifier = Modifier.fillMaxWidth(),
-                        )
-                    },
-                    backgroundColor = Theme.colors.primary,
-                    contentColor = Theme.colors.onPrimary,
-                )
-            },
+@Composable
+fun LoginScreen(navigator: Navigator, state: LoginState, action: (LoginIntent) -> Unit) {
```

Comment:
Navigator should not be part of the Screen, kills all the state-less point
Rather should be an Action to be handled on the higher lever

Reply:
Fixed